app/models/room.rb

```diff
@@ -6,11 +6,11 @@ class Room < ApplicationRecord # :nodoc:
 
   # Enums
   enum room_type: %i[single double triple family_suite]
-  enum status: %i[available unavailable]
+  enum room_status: %i[available unavailable]
 
   # Validations
   validates :number, presence: true, uniqueness: true
   validates :room_type, presence: true
-  validates :status, presence: true
+  validates :room_status, presence: true
```

Comment:
Same as before, I think these prefixes are redundant. Maybe you could have a boolean `available` flag instead of an enum, if there are only two statuses.

Reply:
Instead of validating `presence`, you should validate that the status is within the list of status enum, using [`inclusion` option](https://guides.rubyonrails.org/v7.0/active_record_validations.html#inclusion)

(Also, because you have already set default values, so you won't get nil on initial creation anyways, so it just comes down to when it is being updated, so then you just need to make sure that the updated value is within the list....and nil is not within the list)

So something like:
`validates :status, inclusion: { in: Room.status.keys }`

Should also do that in Reservation enum. 

Fun fact: in rails 7.1, a new 'validate` option is added to `enum`